datalake/src/main/java/com/sequenceiq/datalake/service/sdx/cert/CertRenewalService.java

```diff
@@ -72,22 +80,40 @@ public FlowIdentifier triggerInternalRenewCertificate(SdxCluster sdxCluster) {
-                FlowIdentifier flowIdentifier = ThreadBasedUserCrnProvider.doAsInternalActor(
-                        regionAwareInternalCrnGeneratorFactory.iam().getInternalCrnForServiceAsString(),
-                        () -> stackV4Endpoint.renewCertificate(WORKSPACE_ID_DEFAULT, sdxCluster.getClusterName(), userCrn));
-                saveChainIdAndStatus(sdxCluster, flowIdentifier);
+                try {
+                    FlowIdentifier flowIdentifier = ThreadBasedUserCrnProvider.doAsInternalActor(
+                            regionAwareInternalCrnGeneratorFactory.iam().getInternalCrnForServiceAsString(),
+                            () -> stackV4Endpoint.renewCertificate(WORKSPACE_ID_DEFAULT, sdxCluster.getClusterName(), userCrn));
+                    saveChainIdAndStatus(sdxCluster, flowIdentifier);
+                } catch (ClientErrorException e) {
+                    try {
+                        throw new BadRequestException("Renew certificate failed: " + e.getResponse().readEntity(ExceptionResponse.class).getMessage(), e);
+                    } catch (ProcessingException processingException) {
+                        LOGGER.warn("Can't read response", processingException);
+                        throw e;
+                    }
+                }
             });
         } catch (TransactionExecutionException e) {
-            throw new TransactionRuntimeExecutionException(e);
+            throw e.getCause();
         }
     }
 
     public void renewInternalCertificate(SdxCluster sdxCluster) {
         try {
-            FlowIdentifier flowIdentifier = ThreadBasedUserCrnProvider.doAsInternalActor(
-                    regionAwareInternalCrnGeneratorFactory.iam().getInternalCrnForServiceAsString(),
-                    () -> stackV4Endpoint.renewInternalCertificate(WORKSPACE_ID_DEFAULT, sdxCluster.getStackCrn()));
-            transactionService.required(() -> saveChainIdAndStatus(sdxCluster, flowIdentifier));
+            try {
+                FlowIdentifier flowIdentifier = ThreadBasedUserCrnProvider.doAsInternalActor(
+                        regionAwareInternalCrnGeneratorFactory.iam().getInternalCrnForServiceAsString(),
+                        () -> stackV4Endpoint.renewInternalCertificate(WORKSPACE_ID_DEFAULT, sdxCluster.getStackCrn()));
+                transactionService.required(() -> saveChainIdAndStatus(sdxCluster, flowIdentifier));
+            } catch (ClientErrorException e) {
+                try {
+                    throw new BadRequestException("Renew internal certificate failed: " + e.getResponse().readEntity(ExceptionResponse.class).getMessage(), e);
+                } catch (ProcessingException processingException) {
+                    LOGGER.warn("Can't read response", processingException);
+                    throw e;
+                }
+            }
         } catch (TransactionExecutionException e) {
             throw new TransactionRuntimeExecutionException(e);
```

Comment:
Should this not also throw the cause like the non-internal version?

Reply:
good catch, thanks!